### ISSUE TITLE
Failing to install on Docker container

### ISSUE THREAD

**Issue by fagnercarvalho** [racer-rust/racer#1171]
I'm inside a Docker container and I'm not able to install racer, any idea of what could be happening?

```
rustup toolchain add nightly
cargo +nightly install racer
```

Error:
```
error[E0199]: implementing the trait `Step` is not unsafe
  --> /cargo/registry/src/github.com-1ecc6299db9ec823/rustc-ap-rustc_span-718.0.0/src/def_id.rs:12:1
   |
12 | / rustc_index::newtype_index! {
13 | |     pub struct CrateId {
14 | |         ENCODABLE = custom
15 | |     }
16 | | }
   | |_^
   |
   = note: this error originates in the macro `$crate::newtype_index` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0199]: implementing the trait `Step` is not unsafe
   --> /cargo/registry/src/github.com-1ecc6299db9ec823/rustc-ap-rustc_span-718.0.0/src/def_id.rs:208:1
    |
208 | / rustc_index::newtype_index! {
209 | |     /// A DefIndex is an index into the hir-map for a crate, identifying a
210 | |     /// particular definition. It should really be considered an interned
211 | |     /// shorthand for a particular DefPath.
...   |
219 | |     }
220 | | }
    | |_^
    |
    = note: this error originates in the macro `$crate::newtype_index` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0199]: implementing the trait `Step` is not unsafe
    --> /cargo/registry/src/github.com-1ecc6299db9ec823/rustc-ap-rustc_span-718.0.0/src/symbol.rs:1510:1
     |
1510 | / rustc_index::newtype_index! {
1511 | |     pub struct SymbolIndex { .. }
1512 | | }
     | |_^
     |
     = note: this error originates in the macro `$crate::newtype_index` (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0199`.
error: could not compile `rustc-ap-rustc_span`

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: failed to compile `racer v2.1.47`, intermediate artifacts can be found at `/tmp/cargo-installae0NEO`

Caused by:
  build failed
```

**Comment by fagnercarvalho** [comment]
I guess #1170 should fix this, correct?

**Comment by fagnercarvalho** [comment]
Fixed by #1170, closing.
